crates/sui-core/src/consensus_types/consensus_output_api.rs

```diff
@@ -89,3 +90,59 @@ impl ConsensusOutputAPI for narwhal_types::ConsensusOutput {
             }).collect()
     }
 }
+
+impl ConsensusOutputAPI for mysticeti_core::consensus::linearizer::CommittedSubDag {
+    fn reputation_score_sorted_desc(&self) -> Option<Vec<(AuthorityIndex, u64)>> {
+        // TODO: Implement this in Mysticeti.
+        None
+    }
+
+    fn leader_round(&self) -> u64 {
+        self.anchor.round
+    }
+
+    fn leader_author_index(&self) -> AuthorityIndex {
+        self.anchor.authority as AuthorityIndex
+    }
+
+    fn commit_timestamp_ms(&self) -> u64 {
+        // TODO: Enforce ordered timestamp in Mysticeti
+        0
+    }
+
+    fn commit_sub_dag_index(&self) -> u64 {
+        // TODO: Mysticeti doesn't have this concept right now
+        self.anchor.round
+    }
+
+    fn into_transactions(self) -> ConsensusOutputTransactions {
+        self.blocks
+            .into_iter()
+            .map(|block| {
+                let round = block.round();
+                let author = block.author() as AuthorityIndex;
+                let transactions: Vec<_> = block
+                    .statements()
+                    .iter()
+                    .enumerate()
+                    .flat_map(|(idx, stmt)| match stmt {
+                        BaseStatement::Share(tx) => {
```

Comment:
You can also use `StatementBlock::shared_transactions` which gives you iterator of transactions